shared/chat/conversation/input/index.js.flow

```diff
@@ -28,4 +28,25 @@ export type Props = {
   typing: Array<Types.Username>,
 }
 
+export type MentionHocProps = {
+  insertMention: (u: string, options?: {notUser: boolean}) => void,
+  switchMention: (u: string) => void,
+  insertChannelMention: (c: string, options?: {notChannel: boolean}) => void,
+  switchChannelMention: (c: string) => void,
+  onKeyDown: (e: SyntheticKeyboardEvent<>) => void,
+  onKeyUp: (e: SyntheticKeyboardEvent<*>) => void,
+  onEnterKeyDown: (e: SyntheticKeyboardEvent<>) => void,
+  upArrowCounter: number,
```

Comment:
May be worth decomposing these counters into a separate type here, so that we have one place to update for them (instead of here and `MentionHocState`)